source/setup/components/ArchiveUnlockPage.js

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { Input as ButtercupInput, Button as ButtercupButton } from "@buttercup/ui";
+import LayoutMain from "./LayoutMain.js";
+
+const PasswordRow = styled.div`
+    width: 100%;
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-start;
+    align-items: center;
+    margin-bottom: 30px;
+`;
+const PasswordLabel = styled.label`
+    margin-right: 6px;
+`;
+const ButtonsRow = styled.div`
+    width: 100%;
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-end;
+    align-items: center;
+
+    > * {
+        margin-left: 12px;
+    }
+`;
+
+class ArchiveUnlockPage extends Component {
+    static propTypes = {
+        archiveTitle: PropTypes.string.isRequired,
+        isEditing: PropTypes.bool.isRequired,
+        onLockArchive: PropTypes.func.isRequired,
+        onRemoveArchive: PropTypes.func.isRequired,
+        onUnlockArchive: PropTypes.func.isRequired,
+        sourceID: PropTypes.string.isRequired,
+        state: PropTypes.oneOf(["locked", "unlocked"]).isRequired
+    };
+
+    constructor(props) {
+        super(props);
+        // We store some details in the state, because they're sensitive:
+        this.state = {
+            masterPassword: ""
+        };
+    }
+
+    componentDidMount() {
+        setTimeout(() => {
```

Comment:
timeout is not usually needed as its always immediately available, but maybe you're doing it for UX

Reply:
Yeah it's for the UX.. the input won't focus if I attempt to do it immediately (react state breaks focusing so `autofocus` isn't available either).